pkg/sensors/program/loader_windows.go

```diff
@@ -0,0 +1,480 @@
+			if err != nil {
+				return nil, err
+			}
+
+			unloader = append(unloader, un)
+		}
+		return unloader, nil
+	}
+}
+
+func LoadKprobeProgramAttachMany(bpfDir string, load *Program, syms []string, verbose int) error {
+	opts := &LoadOpts{
+		Attach: KprobeAttachMany(load, syms, bpfDir),
+	}
+	return loadProgram(bpfDir, load, opts, verbose)
+}
+
+func LoadUprobeProgram(bpfDir string, load *Program, verbose int) error {
+	opts := &LoadOpts{
+		Attach: UprobeAttach(load),
+	}
+	return loadProgram(bpfDir, load, opts, verbose)
+}
+
+func LoadMultiKprobeProgram(bpfDir string, load *Program, verbose int) error {
+	opts := &LoadOpts{
+		Attach: MultiKprobeAttach(load, bpfDir),
+		Open:   KprobeOpen(load),
+	}
+	return loadProgram(bpfDir, load, opts, verbose)
+}
+
+func LoadFmodRetProgram(bpfDir string, load *Program, progName string, verbose int) error {
+	return fmt.Errorf("not supported on windows")
+}
+
+func LoadTracingProgram(bpfDir string, load *Program, verbose int) error {
+	opts := &LoadOpts{
+		Attach: TracingAttach(),
+	}
+	return loadProgram(bpfDir, load, opts, verbose)
+}
+
+func LoadLSMProgram(bpfDir string, load *Program, verbose int) error {
+	opts := &LoadOpts{
+		Attach: LSMAttach(),
+		Open:   LSMOpen(load),
+	}
+	return loadProgram(bpfDir, load, opts, verbose)
+}
+
+func LoadLSMProgramSimple(bpfDir string, load *Program, verbose int) error {
+	opts := &LoadOpts{
+		Attach: LSMAttach(),
+	}
+	return loadProgram(bpfDir, load, opts, verbose)
+}
+
+func LoadMultiUprobeProgram(bpfDir string, load *Program, verbose int) error {
+	opts := &LoadOpts{
+		Attach: MultiUprobeAttach(load),
+	}
+	return loadProgram(bpfDir, load, opts, verbose)
+}
+
+// MissingConstantsError is returned by [rewriteConstants].
+type MissingConstantsError struct {
+	// The constants missing from .rodata.
+	Constants []string
+}
+
+func (m *MissingConstantsError) Error() string {
+	return fmt.Sprintf("some constants are missing from .rodata: %s", strings.Join(m.Constants, ", "))
+}
+
+func doLoadProgram(
+	bpfDir string,
+	load *Program,
+	loadOpts *LoadOpts,
+	verbose int,
+) (*LoadedCollection, error) {
+
+	coll, err := ebpf.LoadCollection(load.Name)
+	if err != nil {
+		logger.GetLogger().WithError(err).WithField("Error ", err.Error()).Warn(" Failed to load Native Windows Collection ")
+		return nil, err
+	}
+	bpf.SetExecCollection(coll)
+
+	collMaps := map[ebpf.MapID]*ebpf.Map{}
+	// we need a mapping by ID
+	for _, m := range coll.Maps {
+
+		info, err := m.Info()
+		if err != nil {
+			logger.GetLogger().WithError(err).WithField("map", m.String()).Warn("failed to retrieve BPF map info")
+			continue
+		}
+		id, available := info.ID()
+		if !available {
+			logger.GetLogger().WithField("map", m.String()).Warn("failed to retrieve BPF map ID, you might be running <4.13")
+			continue
+		}
+		collMaps[id] = m
+
+		//ToDo: Uncomment after map pinning issue is fixed
```

Comment:
also linkPin function is removed, is that not supported as well?

Reply:
What is the issue so I can look into it?

Reply:
Pinned ringbuffer maps had a problem with out undocumented use of device io. I will verify once the sync ringbuffer API is made available.